plugins/lookup/vmware_moid.py

```diff
@@ -0,0 +1,526 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+lookup: vmware_moid
+short_description: Look up MOID for vSphere objects using vCenter REST API.
+description:
+    - Returns MOID of the object contained in the specified path.
+options:
+    _terms:
+        description: objects to query.
+    vcenter_hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter.
+        required: true
+        type: str
+    vcenter_password:
```

Comment:
Add support for PROXY Parameters as well.

Reply:
Not yet addressed.

Reply:
@goneri what do you think about this?

Reply:
The collection does not support HTTP proxy yet. @Akasurde, I suggest we address this in another PR. This sounds like a solid feature for a 2.1.0 or 2.2.0.